pkg/pgmodel/end_to_end_tests/promql_query_endpoint_test.go

```diff
@@ -272,59 +297,2693 @@ func TestPromQLQueryEndpoint(t *testing.T) {
+			realDataset: true,
+		},
+		{
+			name:        "real query 513",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.1, 0.1)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 514",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.1, 0.5)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 515",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.1, 0.8)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 516",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.5, 0.1)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 517",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.5, 0.5)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 518",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.5, 0.8)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 519",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.8, 0.1)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 520",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.8, 0.5)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 521",
+			query:       `holt_winters(demo_disk_usage_bytes[10m], 0.8, 0.8)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 522",
+			query:       `max_over_time((time() - max(demo_batch_last_success_timestamp_seconds) < 1000)[5m:10s] offset 5m)`,
+			realDataset: true,
+		},
+		{
+			name:        "real query 523",
+			query:       `avg_over_time(rate(demo_cpu_usage_seconds_total[1m])[2m:10s])`,
+			realDataset: true,
+		},
+	}
+
+	withDB(t, *testDatabase, func(db *pgxpool.Pool, t testing.TB) {
+		// Ingest test dataset.
+		dataset := generateLargeTimeseries()
+		if *extendedTest {
+			dataset = append(dataset, generateRealTimeseries()...)
+		}
+		ingestQueryTestDataset(db, t, dataset)
+		// Getting a read-only connection to ensure read path is idempotent.
+		readOnly := testhelpers.GetReadOnlyConnection(t, *testDatabase)
+		defer readOnly.Close()
+
+		var tester *testing.T
+		var ok bool
+		if tester, ok = t.(*testing.T); !ok {
+			t.Fatalf("Cannot run test, not an instance of testing.T")
+			return
+		}
+
+		r := pgmodel.NewPgxReader(readOnly, nil, 100)
+		queryable := query.NewQueryable(r.GetQuerier())
+		queryEngine := query.NewEngine(log.GetLogger(), time.Minute)
+
+		apiConfig := &api.Config{}
+		instantQuery := api.Query(apiConfig, queryEngine, queryable)
+		rangeQuery := api.QueryRange(apiConfig, queryEngine, queryable)
+
+		apiURL := fmt.Sprintf("http://%s:%d/api/v1", testhelpers.PromHost, testhelpers.PromPort.Int())
+		client := &http.Client{Timeout: 10 * time.Second}
+
+		start := time.Unix(startTime/1000, 0)
+		end := time.Unix(endTime/1000, 0)
+		var (
+			requestCases []requestCase
+			req          *http.Request
+			err          error
+		)
+		for _, c := range testCases {
+			if c.realDataset {
```

Comment:
Instead of having this logic where `TestPromQLQueryEndpoint` runs different queries depending on *extendedTest or not. I'd prefer a `TestPromQLQueryEndpointGeneratedDataset` and `TestPromQLQueryRealDataset' where the latter has t.Skip() called when !*extendendedTest. Both these functions would define their own testCases and then call a common function that would take in testCases and run the appropriate tests. Also start and end times can then be clearly defined for each TestPromQL function